components/PlantCard.js

```diff
@@ -1,22 +1,30 @@
 import styled from "styled-components";
+import PlantFavIcon from "@/public/Icons/plant-line.svg";
+import PlantFavIconFill from "@/public/Icons/plant-fill.svg";
+import Link from "next/link";
 
-export default function PlantCard({ plant }) {
+export default function PlantCard({ plant, toggleFavourite }) {
   return (
     <Container>
+      <button
+        aria-label={
+          plant.isFavourite ? "Remove from favorites" : "Add to favorites"
+        }
+        onClick={() => toggleFavourite(plant.id)}
+      >
+        {plant.isFavourite ? <PlantFavIconFill /> : <PlantFavIcon />}
+      </button>
       <h2>{plant.name}</h2>
       <StyledParagraph>{plant.botanicalName}</StyledParagraph>
 
-      <RoundImage
-        alt={`Image of ${plant.name}`}
-        src={plant.imageUrl}
-      ></RoundImage>
+      <RoundImage alt={`Image of ${plant.name}`} src={plant.imageUrl} />
 
       <StyledLink href={`/${plant.id}`}>see more</StyledLink>
     </Container>
   );
 }
 
-const StyledLink = styled.a`
+const StyledLink = styled(Link)`
```

Comment:
Have a look at your RoundImage styled component: You should always make use of the Next Image Component instead of an `<img>` tag.